.github/workflows/ci.yml

```
@@ -0,0 +1,180 @@
+name: CI - Lint and Build
+
+on:
+  pull_request:
+    branches: [ main ] # Or your main branch name
+
+jobs:
+  lint-and-build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
```

actionlint (1.7.4)

the runner of "actions/checkout@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)